hexl/ntt/ntt-radix-2.cpp

```diff
@@ -34,7 +36,88 @@ void ForwardTransformToBitReverseRadix2(
   uint64_t twice_modulus = modulus << 1;
   size_t t = (n >> 1);
 
-  for (size_t m = 1; m < n; m <<= 1) {
+  // In case of out-of-place operation do first pass and convert to in-place
```

Comment:
This adds a lot of extra code.  Is it possible to roll this back into the for-loop? (same with the radix-4 case) Something like the below may work:
```
for (size_t m = 2; m < n; m <<= 1) {
    uint64_t* X_r = result + j1;
    uint64_t* Y_r = X_r + t;
    const uint64_t* X_op = X_r;
    const uint64_t* Y_op = Y_r;
    if (m == 1) {
      X_op = operand;
      Y_op = X_op + t;
    }
...
}
```

Reply:
I have done this and I think there is a performance hit.
[InLoopVsOutOfLoop.xlsx](https://github.com/intel/hexl/files/7295064/InLoopVsOutOfLoop.xlsx)
However, I don't understand the improvement on BM_FwdNTTNativeRadix4/4096